components/eamxx/src/share/field/field_manager.cpp

```diff
@@ -516,88 +506,109 @@ void FieldManager::registration_ends ()
         }
       }
 
-      // Figure out the layout of the fields in this cluster,
-      // and make sure they all have the same layout
-      LayoutType lt = LayoutType::Invalid;
-      FieldLayout f_layout = FieldLayout::invalid();
-      for (const auto& fname : cluster_ordered_fields) {
-        const auto& f = m_fields.at(fname);
-        const auto& id = f->get_header().get_identifier();
-        if (lt==LayoutType::Invalid) {
-         f_layout = id.get_layout();
-         lt = f_layout.type();
-        } else {
-          EKAT_REQUIRE_MSG (lt==id.get_layout().type(),
-              "Error! Found a group to bundle containing fields with different layouts.\n"
-              "       Group name: " + cluster_name + "\n"
-              "       Layout 1: " + e2str(lt) + "\n"
-              "       Layout 2: " + e2str(id.get_layout().type()) + "\n");
+      // We must allocate this cluster field on every grid that
+      // exists in the group requests contained in the cluster
```

Comment:
We now allocate each cluster on every grid that requests a group in the cluster.